examples/mortality.py

```diff
@@ -0,0 +1,276 @@
+            "beta_age_drift", dist.Normal(0, beta_age_drift_scale)
+        )
+        beta_age = jnp.cumsum(beta_age_drift, -2)
+
+    # age-space interactions
+    with age_plate, space_plate:
+        xi = numpyro.sample("xi", dist.Normal(alpha_age + alpha_s2, sigma_xi))
+
+    # space-time random walk
+    with space_plate, year_plate:
+        nu_drift = numpyro.sample("nu_drift", dist.Normal(beta_s2, sigma_nu))
+        nu = jnp.pad(jnp.cumsum(nu_drift, -1), [(0, 0), (1, 0)])
+
+    # age-time random walk
+    with age_plate, year_plate:
+        gamma_drift = numpyro.sample("gamma_drift", dist.Normal(beta_age, sigma_gamma))
+        gamma = jnp.pad(jnp.cumsum(gamma_drift, -1), [(0, 0), (0, 0), (1, 0)])
+
+    # likelihood
+    latent_rate = xi + nu + gamma
+    with numpyro.plate("N", N):
+        mu_logit = latent_rate[space, age, time]
+        mu = numpyro.deterministic("mu", expit(mu_logit))
+        numpyro.sample(
+            "deaths",
+            dist.BetaBinomial(mu * theta, (1 - mu) * theta, population),
+            obs=deaths,
+        )
+
+
+def print_model_shape(model, age, space, time, lookup, population):
+    with numpyro.handlers.seed(rng_seed=1):
+        trace = numpyro.handlers.trace(model).get_trace(
+            age=age,
+            space=space,
+            time=time,
+            lookup=lookup,
+            population=population,
+        )
+    print(numpyro.util.format_shapes(trace))
+
+
+def run_inference(model, age, space, time, lookup, population, deaths, rng_key, args):
+    kernel = NUTS(model, init_strategy=init_to_median)
+    mcmc = MCMC(
+        kernel,
+        num_warmup=args.num_warmup,
+        num_samples=args.num_samples,
+        num_chains=args.num_chains,
+        progress_bar=False if "NUMPYRO_SPHINXBUILD" in os.environ else True,
+    )
+    mcmc.run(rng_key, age, space, time, lookup, population, deaths)
+    mcmc.print_summary()
+    return mcmc.get_samples()
+
+
+def main(args):
+    print("Fetching simulated data...")
+    a, s2, t, lookup, population, deaths = get_data()
+
+    print("Model shape:")
+    print_model_shape(model, a, s2, t, lookup, population)
+
+    print("Starting inference...")
+    rng_key = random.PRNGKey(args.rng_seed)
+    run_inference(model, a, s2, t, lookup, population, deaths, rng_key, args)
+
+
+if __name__ == "__main__":
+    assert numpyro.__version__.startswith("0.8.0")
+
+    parser = argparse.ArgumentParser(description="Mortality regression model")
+    parser.add_argument("-n", "--num-samples", nargs="?", default=100, type=int)
+    parser.add_argument("--num-warmup", nargs="?", default=50, type=int)
+    parser.add_argument("--num-chains", nargs="?", default=1, type=int)
+    parser.add_argument("--device", default="cpu", type=str, help='use "cpu" or "gpu".')
+    parser.add_argument(
+        "--rng_seed", default=21, type=int, help="random number generator seed"
+    )
+    args = parser.parse_args()
+
+    numpyro.set_platform(args.device)
```

Comment:
Maybe also set `numpyro.enable_x64()`? I think it is better to work in double precision for this model.

Reply:
Sure I'll add that. Is there a downside to `numpyro.enable_x64()`? Would it eat up a lot more memory? When I run this on the full, real dataset with nimble, it gets very memory-intensive (300+ GB for 4 parallel chains)